pkg/policycache/cache.go

```diff
@@ -5,22 +5,23 @@ import (
 
 	"github.com/go-logr/logr"
 	kyverno "github.com/kyverno/kyverno/pkg/api/kyverno/v1"
+	kyvernolister "github.com/kyverno/kyverno/pkg/client/listers/kyverno/v1"
+	policy2 "github.com/kyverno/kyverno/pkg/policy"
 )
 
 type pMap struct {
 	sync.RWMutex
-	// dataMap field stores ClusterPolicies
-	dataMap map[PolicyType][]*kyverno.ClusterPolicy
-	// nsDataMap field stores Namespaced Policies for each namespaces.
-	// The Policy is converted internally to ClusterPolicy and stored as a ClusterPolicy
-	// Since both the policy use same type (i.e. Policy), Both policies can be differentiated based on
-	// "Kind" or "namespace". When the Policy is converted it will retain the value of kind as "Policy".
-	// Cluster policy will be having namespace as Blank (""), but Policy will always be having namespace field and "default" value by default
-	nsDataMap map[string]map[PolicyType][]*kyverno.ClusterPolicy
+
+	kindDataMap map[string]map[PolicyType][]string
 
 	// nameCacheMap stores the names of all existing policies in dataMap
 	// Policy names are stored as <namespace>/<name>
 	nameCacheMap map[PolicyType]map[string]bool
+
+	pLister kyvernolister.ClusterPolicyLister
+
+	// npLister can list/get namespace policy from the shared informer's store
+	npLister kyvernolister.PolicyLister
```

Comment:
Can we embed these 2 listers into `policyCache` and leave `pMap` as the fundamental map? They are thread-safe, so we don't need to add the lock to them. Otherwise, the performance can be compromised.